packages/vm/lib/evm/interpreter.ts

```diff
@@ -193,6 +197,29 @@ export default class Interpreter {
       memoryWordCount: this._runState.memoryWordCount,
       codeAddress: this._eei._env.codeAddress,
     }
+
+    // Create opTrace for debug functionality
+    let hexStack = []
+    hexStack = eventObj.stack.map((item: any) => {
+      return '0x' + new BN(item).toString(16, 0)
+    })
+
+    const name = eventObj.opcode.name
+    const nameLC = name.toLowerCase()
+    const opTrace = {
+      pc: eventObj.pc,
+      op: name,
+      gas: '0x' + eventObj.gasLeft.toString('hex'),
+      gasCost: '0x' + eventObj.opcode.fee.toString(16),
+      stack: hexStack,
+      depth: eventObj.depth,
+    }
+
+    if (!(nameLC in this.opDebuggers)) {
+      this.opDebuggers[nameLC] = createDebugLogger(`vm:ops:${nameLC}`)
+    }
+    this.opDebuggers[nameLC](JSON.stringify(opTrace))
```

Comment:
This is really nice.

@jochem-brouwer You mentioned wanting to diff the geth / ethereumjs traces - would the most convenient thing be to have two files with JSONs of the respective traces? 

If so debug has [custom formatters][1] which might help with this. 

I think you could collect the steps into a trace and then write to `ejs.vm.<txhash>.json`, when `DEBUG=vm:savetrace` (or something)

[1]: https://www.npmjs.com/package/debug#custom-formatters

Reply:
I would guess this is a good suggestion since comparing might be easiest on a per file basis with some linux CLI tools? @jochem-brouwer WDYT?